src/AppInstallerRepositoryCore/Microsoft/SQLiteIndexSource.cpp

```diff
@@ -153,14 +206,48 @@ namespace AppInstaller::Repository::Microsoft
             {
                 return false;
             }
+        };
 
-        private:
-            SQLiteIndex::IdType m_idId;
+        // The IPackage impl for SQLiteIndexSource of Installed packages.
+        struct InstalledPackage : public PackageBase, public IPackage
+        {
+            using PackageBase::PackageBase;
+
+            // Inherited via IPackage
+            Utility::LocIndString GetProperty(PackageProperty property) const override
+            {
+                return PackageBase::GetProperty(property);
+            }
+
+            std::shared_ptr<IPackageVersion> GetInstalledVersion() const override
+            {
+                return GetLatestVersionInternal();
+            }
+
+            std::vector<PackageVersionKey> GetAvailableVersionKeys() const override
+            {
+                return {};
+            }
+
+            std::shared_ptr<IPackageVersion> GetLatestAvailableVersion() const override
+            {
+                return {};
+            }
+
+            std::shared_ptr<IPackageVersion> GetAvailableVersion(const PackageVersionKey&) const override
+            {
+                return {};
+            }
+
+            bool IsUpdateAvailable() const override
+            {
+                return false;
+            }
         };
     }
 
-    SQLiteIndexSource::SQLiteIndexSource(const SourceDetails& details, std::string identifier, SQLiteIndex&& index, Synchronization::CrossProcessReaderWriteLock&& lock) :
-        m_details(details), m_identifier(std::move(identifier)), m_lock(std::move(lock)), m_index(std::move(index))
+    SQLiteIndexSource::SQLiteIndexSource(const SourceDetails& details, std::string identifier, SQLiteIndex&& index, Synchronization::CrossProcessReaderWriteLock&& lock, bool isInstalledSource) :
+        m_details(details), m_identifier(std::move(identifier)), m_lock(std::move(lock)), m_isInstalled(isInstalledSource), m_index(std::move(index))

```

Comment:
You might have answered this before, should this be just in the header?

Reply:
I prefer non-trivial functions, including constructors, to be defined in the cpp.

---
In reply to: [507279645](https://github.com/microsoft/winget-cli/pull/598#discussion_r507279645) [](ancestors = 507279645)